interp/src/interpret_control.rs

```diff
@@ -0,0 +1,171 @@
+//! Inteprets a control in a component.
+
+use super::{
+    environment::Environment, interpret_group::GroupInterpreter, interpreter,
+};
+use calyx::{
+    errors::{Error, FutilResult},
+    ir,
+};
+use std::rc::Rc;
+
+/// Interpret a control.
+/// TODO: Attributes, implementation for some component variants
+pub struct ControlInterpreter {
+    /// The environment
+    pub environment: Environment,
+
+    /// The component the control belongs to
+    // XX(2/25 meeting): we might not need this? all the information is in the IR
+    pub component: String,
+
+    /// The control to interpret
+    pub control: ir::RRC<ir::Control>,
+}
+
+impl ControlInterpreter {
+    /// Construct ControlInterpreter
+    /// env : Initial environment
+    /// comp : Name of the component the Control is from
+    /// ctrl : The control to interpret
+    pub fn init(
+        env: Environment,
+        comp: String,
+        ctrl: ir::RRC<ir::Control>,
+    ) -> Self {
+        Self {
+            environment: env,
+            component: comp,
+            control: ctrl,
+        }
+    }
+
+    /// Interpret this control
+    pub fn interpret(self) -> FutilResult<Environment> {
+        eval_control(
+            &(*self.control.borrow()),
+            self.component.clone(),
+            self.environment,
+        )
+    }
+}
+
+/// Helper function to evaluate control
+fn eval_control(
+    ctrl: &ir::Control,
+    comp: String,
+    env: Environment,
```

Comment:
This is why it doesn't seem like the interpreter should really own the environment, since `interpret()` just passes in the stored environment directly